### PR TITLE
fix(refine): fix get one return type

### PR DESCRIPTION
**Pull request by self-transition**
- Fix get one return type
- Remove redundant eslint file

In the previous implementation, when the getOne method didn't get the right value, it returned the consistent type of the 「Unstructured」 structure, meanwhile its id property was set by the empty string.  
But the expected behavior is to return `null`.

### CODE REVIEW

File: src/data-provider/index.ts [hunk 1]
```diff
@@ -41,16 +41,18 @@ export const dataProvider = (
...
     const { items, kind, apiVersion } = await globalStore.get(resource, meta);
-
     const data = items.find(
       item =>
         item.metadata.name === name && item.metadata.namespace === namespace
     );
-
+    if (!data) {
+      console.error(`resource: ${resource} not include id: ${id}`)
+      return { data: null as unknown as TData }
```
Comment (huangjunjia) [inline src/data-provider/index.ts:50]: Why not add type `null` to the return types, but specify it as the type `TData`?

Comment (self-transition) [reply src/data-provider/index.ts:50]: Limited by https://github.com/refinedev/refine/blob/68edee93080122b0e5824dc6830232be059afa6f/packages/core/src/contexts/data/IDataContext.ts#L274

Comment (huangjunjia) [reply src/data-provider/index.ts:50]: Okey then.

**Review by huangjunjia — APPROVED** [review]
LGTM

**Pull request merged**